lib/modules/auth/widgets/login.widget.dart

```diff
@@ -66,81 +45,93 @@ class LoginState extends State<Login> {
-                  label: Text('Password'), prefixIcon: Icon(Icons.password)),
+                label: Text(S.of(context).LoginScreen_password_input_label),
+              ),
             ),
             SizedBox(height: ThemeConfig.of(context)!.largestSpacing),
             SizedBox(
               child: ElevatedButton(
-                onPressed: () {
-                  // Validate returns true if the form is valid, or false otherwise.
-                  if (_formKey.currentState!.validate()) {
-                    login();
-                  }
-                },
-                child: const Text('Login'),
+                onPressed: () => login(authProvider),
+                child: Text(S.of(context).LoginScreen_login_button),
               ),
             ),
             SizedBox(height: ThemeConfig.of(context)!.largeSpacing),
-            SizedBox(
-                child: Row(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: [
-                Expanded(child: Divider()),
-                Padding(
-                  padding: const EdgeInsets.symmetric(horizontal: 10),
-                  child: Text('OR'),
-                ),
-                Expanded(child: Divider())
-              ],
-            )),
+            LabeledDivider(
+              label: Text(S.of(context).LoginScreen_or),
+            ),
             SizedBox(height: ThemeConfig.of(context)!.largeSpacing),
-            SizedBox(
-              child: ElevatedButton(
-                onPressed: null,
-                child: const Text('Register'),
-              ),
+            OutlinedButton(
+              onPressed: null,
+              child: Text(S.of(context).LoginScreen_register_button),
             ),
             SizedBox(height: ThemeConfig.of(context)!.mediumSpacing),
-            SizedBox(
-              child: ElevatedButton(
-                onPressed: null,
-                child: const Text('Continue with Google'),
-              ),
+            ElevatedButton(
+              onPressed: null,
+              child: Text(S.of(context).LoginScreen_google_button),
             ),
             SizedBox(height: ThemeConfig.of(context)!.mediumSpacing),
-            SizedBox(
-              child: ElevatedButton(
-                onPressed: null,
-                child: const Text('Continue with Facebook'),
-              ),
+            ElevatedButton(
+              onPressed: null,
+              child: Text(S.of(context).LoginScreen_facebook_button),
             ),
           ],
         ),
       ),
     );
   }
+
+  void login(AuthProvider authProvider) {
+    if (_formKey.currentState?.validate() ?? false) {
+      authProvider
+          .login(emailController.value.text, passwordController.value.text)
+          .whenComplete(() {
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            backgroundColor: authProvider.error == null
+                ? ThemeConfig.of(context)!.successColor
```

Comment:
context ii accesibil inafara metodei build?

Reply:
In StatessWidgets ii accesibil doar in build.
In StafefulWidgets ii accesibil in toata clasa de State.